c7n/resources/iam.py

```diff
@@ -817,6 +817,44 @@ def process(self, resources):
                     pass
 
 
+@Role.action_registry.register('delete')
+class RoleDelete(BaseAction):
+    """Delete an IAM Role.
+
+    For example, if you want to automatically delete an unused IAM role.
+
+    :example:
+
+      .. code-block:: yaml
+
+        - name: iam-delete-unused-role
+          resource: iam-role
+          filters:
+            - type: usage
+              match-operator: all
+              LastAuthenticated: null
+          actions:
+            - delete
+
+    """
+    schema = type_schema('delete')
+    permissions = ('iam:DeleteRole',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('iam')
+        for r in resources:
+            try:
+                client.delete_role(RoleName=r['RoleName'])
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'DeleteConflict':
+                    self.log.warning(
+                        "Cannot delete entity, must remove roles from instance profile first")
+                    continue
+                if e.response['Error']['Code'] == 'NoSuchEntity':
+                    continue
+                raise
```

Comment:
@kapilt If the exception is DeleteConflictException or NoSuchEntityException, don't think it'll ever hit raise in this case. Do you think we should remove it?